src/metadata.py

```diff
@@ -1,50 +1,52 @@
 import json
-import redis
 from abc import ABC, abstractmethod
-from typing import Any, Dict, List, Optional
+from db import Database
+from typing import Any, Dict, Optional
 
 DEFAULT_CACHE_EXPIRE_TIME = 60 * 60 * 12
 
 Metadata = Dict[str, Any]
+MetadataPluginConfig = Dict[str, str]
 
 
 class MetadataPlugin(ABC):
-    #: List of plugin identifiers that plugin depends on
-    __depends_on__: List[str] = []
     #: Enables cache for extracted metadata
     __cacheable__: bool = False
     #: Overrides default cache expire time
     __cache_expire_time__: int = DEFAULT_CACHE_EXPIRE_TIME
+    #: Configuration keys required by plugin with description as a value
+    __config_fields__: Dict[str, str] = {}
```

Comment:
```suggestion
    CONFIG_FIELDS: Dict[str, str] = {}
```

I probably should've complained earlier, but I don't think we should use names with underscores on both sides, because they're reserved.

https://dbader.org/blog/meaning-of-underscores-in-python

```
However, names that have both leading and trailing double underscores are reserved for special use in the language. This rule covers things like __init__ for object constructors, or __call__ to make an object callable.

These dunder methods are often referred to as magic methods—but many people in the Python community, including myself, don’t like that.

It’s best to stay away from using names that start and end with double underscores (“dunders”) in your own programs to avoid collisions with future changes to the Python language.
```

Also, to quote something more normative, PEP8:

```
__double_leading_and_trailing_underscore__: "magic" objects or attributes that live in user-controlled namespaces. E.g. __init__, __import__ or __file__. Never invent such names; only use them as documented.
```

(note that forward underscores only will apply name mangling, so they may not be what is intended here)

Reply:
Yup, that was convention initiated before. It's good time to refactor that.

Reply:
But I think it should be still lowercase, it's rather class attribute than constant.